controllers/managedfusionoffering_controller.go

```diff
@@ -124,17 +131,43 @@ func (r *ManagedFusionOfferingReconciler) reconcilePhases() (reconcile.Result, e
 
 		} else if ready {
 			if removed := utils.RemoveFinalizer(r.managedFusionOffering, managedFusionFinalizer); removed {
-				r.Log.Info(fmt.Sprintf("removing finalizer from %s secret", managedFusionSecretName))
+				r.Log.Info(fmt.Sprintf(
+					"removing finalizer from %s (%s)",
+					r.managedFusionOffering.Name,
+					r.managedFusionOffering.Namespace,
+				))
 				if err := r.update(r.managedFusionOffering); err != nil {
-					return ctrl.Result{}, fmt.Errorf("failed to remove finalizer from %s secret: %v", managedFusionSecretName, err)
+					return ctrl.Result{}, fmt.Errorf(
+						"failed to remove finalizer from %s (%s): %v",
+						r.managedFusionOffering.Name,
+						r.managedFusionOffering.Namespace,
+						err,
+					)
 				}
 			}
+			r.Log.Info(fmt.Sprintf("issuing a delete to %s namespace", r.namespace))
+			if err := r.delete(&corev1.Namespace{
```

Comment:
Are we sure it will be ok to delete the ns even before we are sure the offering CR got deleted?

Reply:
I think it is ok to delete the ns before we are sure that offering CR is deleted because control removal of ManagedFusionOffering CR based on finalizer. And once the finalizer is removed, the assumption is that nothing is blocking the removal of the namespace.